### PR TITLE
Fix virtual thread test setup for jdk 20

### PR DESCRIPTION
**Pull request by EricYangIBM**
Run the test (and with --enable-preview) for all java versions, not just
java 19.

Signed-off-by: Eric Yang <eric.yang@ibm.com>

### CODE REVIEW

**Comment by EricYangIBM** [comment]
@tajila

File: test/functional/Java19andUp/build.xml [hunk 1]
```diff
@@ -50,7 +50,7 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 		<echo>===destdir:                      ${DEST}</echo>
 
 		<if>
-			<equals arg1="${JDK_VERSION}" arg2="19"/>
+			<matches string="${JDK_VERSION}" pattern="^(19|20)$$"/>
```
Comment (keithc-ca) [inline test/functional/Java19andUp/build.xml:53]: This should be more general; we shouldn't need to revisit this for jdk21, jdk22, etc. See lines 84-86 for a better condition.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: test/functional/Java19andUp/build.xml (hunk 1 above)
Comment (EricYangIBM) [reply test/functional/Java19andUp/build.xml:53]: The rationale behind this is that 19 and 20 have Loom as a preview but future versions may not. @tajila for clarification
Also the test won't be compiled for versions < 19

Comment (keithc-ca) [reply test/functional/Java19andUp/build.xml:53]: That condition could be condensed to
```
	<not>
		<matches string="${JDK_VERSION}" pattern="^([0-9]|1[0-8])$$" />
	</not>
```

Comment (tajila) [reply test/functional/Java19andUp/build.xml:53]: The features that we are testing in 19 are preview for JDK19 and JDK20 (according to the current code base) but they will not remain preview indefinitely. Its possible that by jdk21 the preview flag will not be needed anymore.

Comment (keithc-ca) [reply test/functional/Java19andUp/build.xml:53]: If we seek to control the use of the `--enable-preview` option, then it should be just that that is version sensitive; the `<javac>` element should be unconditional.

Comment (EricYangIBM) [reply test/functional/Java19andUp/build.xml:53]: > --enable-preview option, then it should be just that that is version sensitive

I seem to recall that we can't do conditionals inside the javac blocks

Comment (keithc-ca) [reply test/functional/Java19andUp/build.xml:53]: That's right, but a `<condition>` element can be used to define the value of the `line` attribute of the `<compilerarg>` element.

Comment (EricYangIBM) [reply test/functional/Java19andUp/build.xml]: Updated

File: test/functional/Java19andUp/build.xml [hunk 2]
```diff
@@ -36,6 +36,17 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
+		<matches string="${JDK_VERSION}" pattern="^(19|20)$$"/>
+		<then>
+			<!-- Loom is a preview feature in Java 19 and 20. -->
+			<property name="preview" value="--enable-preview --source ${JDK_VERSION}" />
+		</then>
+		<else>
+			<property name="preview" value="" />
+		</else>
+	</if>
+
```
Comment (keithc-ca) [inline test/functional/Java19andUp/build.xml:49]: On further reflection, I don't see a problem using `--enable-preview` unconditionally. Even after Loom ceases to be a preview feature, compiling with `--enable-preview` should continue to be successful.

Also, besides being more compact, I suggest we should avoid using ant-contrib features unnecessarily, so if we **_needed_** to have a conditional property value, it should be written in this form:
```
		<condition property="preview" value="--enable-preview" else="">
			<matches string="${JDK_VERSION}" pattern="^(19|20)$$" />
		</condition>
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Comment by keithc-ca** [comment]
jenkins test sanity alinux64 jdknext

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**